code-challenges.test.js

```diff
@@ -10,58 +10,211 @@
+  // Use Object.values() method and pass in object as parameter.
+    // Chain .sort() method to Object.values() method
+      // The .sort() method is made to work with strings. Since we are dealing with numbers, we will need to make the .sort() method setup for sorting numbers. You do that by passing in the function (a, b) => a - b  in the .sort() method.
+// Return the new array.
+
+const mySorter = (object) => {
+  let newArray = Object.values(object).sort((a, b) => a - b)
+  return newArray
+}
+
+// Test Suites: 1 failed, 1 total
+// Tests:       1 failed, 1 passed, 2 total
+
+
+
+
 // --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const accountTransactions1 = [100, -17, -23, -9]
-// Expected output: [100, 83, 60, 51]
+// const accountTransactions1 = [100, -17, -23, -9]
+// // Expected output: [100, 83, 60, 51]
 
-const accountTransactions2 = [250, -89, 100, -96]
-// Expected output: [250, 161, 261, 165]
+// const accountTransactions2 = [250, -89, 100, -96]
+// // Expected output: [250, 161, 261, 165]
+
+// const accountTransactions3 = []
+// // Expected output: []
+
+describe("accum", () => {
+  it("returns an array of the accumulating sum", () => {
+    const accountTransactions1 = [100, -17, -23, -9]
+    const expectedOutput1 = [100, 83, 60, 51]
+    const accountTransactions2 = [250, -89, 100, -96]
+    const expectedOutput2 = [250, 161, 261, 165]
+    const accountTransactions3 = []
+    const expectedOutput3 = []
+    expect(accum(accountTransactions1)).toEqual(expectedOutput1)
+    expect(accum(accountTransactions2)).toEqual(expectedOutput2)
+    expect(accum(accountTransactions3)).toEqual(expectedOutput3)
+  })
+})
+
+// ReferenceError: accum is not defined
 
-const accountTransactions3 = []
-// Expected output: []
 
 // b) Create the function that makes the test pass.
+
+
+// I again spent hours trying this one. I can't figure out the logic. Brain hurts. I have been able to add up all of the items in the array together but not one by one and adding the sum and putting it into a new array.
+
+// Input: array
+// Output: array
+
+// Create a function that takes in an array as parameter
+// Use for loop to iterate over array
+// Return sums of indexes added together 
+
+// const accum = (array) => {
+//   //return array.map((value) => value[i] + value)
+//   let sum = []
+//   for(let i = 0; i < array.length; i++) {
+//     sum.push(array[i] + sum)
+//     //sum = sum + array[i]
+//   }
+//   return sum
+// }
+
+const accum = (array) => {
+  const newArray = []
+  let first = newArray.push(array[0])
+  let second = newArray.push(array[0] + array[1]) 
+  let third = newArray.push(array[0] + array[1] + array[2])
+  let fourth = newArray.push(array[0] + array[1] + array[2] + array[3])
+  return newArray
+}
```

Comment:
The moment we get into figuring out each individual items in an array statically we've usually (although exceptions do exist) gone the wrong way. 

There is a tool that takes in arrays and returns arrays of the same length... may want to use that tool.